src/system/SystemLayerImplSelect.cpp

```diff
@@ -283,6 +325,24 @@ CHIP_ERROR LayerImplSelect::RequestCallbackOnPendingRead(SocketWatchToken token)
             dispatch_activate(watch->mRdSource);
         }
     }
+#elif CHIP_SYSTEM_CONFIG_USE_LIBEV
+    if (mLibEvLoopP == nullptr)
```

Comment:
This makes `CHIP_SYSTEM_CONFIG_USE_LIBEV` mutually exclusive with `CHIP_SYSTEM_CONFIG_USE_DISPATCH`.  But in the Darwin platform manager code the ifdefs are assuming both could be defined at the same time, no?

I think we should stick to them being mutually exclusive and static_asserting that as needed.

Reply:
I agree. I started with `#if`/`#endif` because this gives clearer blocks in diffs and does not touch the existing code. But it was not possible to keep it that way in many places, so sticking with `#if`/`#elif` in all places is probably better.

Reply:
See new commit, now changed to #if/#elif throughout.